packages/stats-py/aqsha_stats/analyses/regresi_logistik.py

```
@@ -0,0 +1,123 @@
+        raise AnalysisError(
+            "invalid_args",
+            f"Variabel terikat '{dependent}' harus biner (2 nilai, mis. 0/1); ditemukan {len(uniques)} nilai unik.",
+        )
+    # Kategori tertinggi = kejadian (1), pola SPSS default (kode terbesar = target).
+    y = (y_raw == uniques[1]).astype(int)
+    X = sm.add_constant(sub[independents].astype(float))
+    try:
+        model = sm.Logit(y, X).fit(disp=0)
+        null = sm.Logit(y, np.ones((len(y), 1))).fit(disp=0)
+    except Exception as exc:  # noqa: BLE001
+        raise AnalysisError("computation_failed", f"Model logistik gagal konvergen: {exc}")
```

Ruff (0.15.20)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)